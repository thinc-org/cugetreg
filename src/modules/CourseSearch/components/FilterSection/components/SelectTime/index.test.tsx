import { shallow } from 'enzyme'

describe('SelectTime', () => {
  const mockOnCheckboxChange = jest.fn()
  const mockOnEndTimeChange = jest.fn()
  const mockOnStartTimeChange = jest.fn()
  const MOCK_START_CHOICES = ['10:30', '11:00', '11:30', '12:00', '12:30']
  const MOCK_ENDTIME_CHOICES = ['11:30', '12:00', '12:30', '13:00', '13:30']
  const MOCK_START_TIME = '11:30'
  const MOCK_END_TIME = '11:30'

  const mockLog = jest.fn()

  const mockUseSelectTime = jest.fn(() => ({
    selectedStartTime: MOCK_START_TIME,
    selectedEndTime: MOCK_END_TIME,
    startTimeChoices: MOCK_START_CHOICES,
    endTimeChoices: MOCK_ENDTIME_CHOICES,
    onStartTimeChange: mockOnStartTimeChange,
    onEndTimeChange: mockOnEndTimeChange,
    checked: true,
    onCheckboxChange: mockOnCheckboxChange,
  }))
  jest.doMock('./hooks/useSelectTime', () => ({ useSelectTime: mockUseSelectTime }))

  it('Should render correctly', async () => {
    const { SelectTime } = await import('.')

    const wrapper = shallow(<SelectTime log={mockLog} />)

    expect(wrapper).toMatchSnapshot()
    // expect(wrapper.f)
  })
})
