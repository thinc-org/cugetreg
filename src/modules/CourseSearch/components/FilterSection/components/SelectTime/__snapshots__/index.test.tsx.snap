// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectTime Should render correctly 1`] = `
<ForwardRef(Stack)>
  <Styled(Component)
    variant="button"
  >
    เวลาเรียน
  </Styled(Component)>
  <ForwardRef(FormControlLabel)
    control={
      <ForwardRef(Checkbox)
        checked={true}
        color="primary"
        name="เวลาเรียน"
        onChange={[MockFunction]}
        sx={
          Object {
            "fontFamily": "Prompt",
          }
        }
      />
    }
    key="periodRange"
    label={
      <ForwardRef(Box)>
        <ForwardRef(Box)
          alignItems="center"
          display="flex"
          justifyContent="space-between"
          mb={2}
        >
          <ForwardRef(Typography)
            mr={2}
            variant="subtitle1"
          >
            ในช่วง
          </ForwardRef(Typography)>
          <ForwardRef(Select)
            disabled={false}
            name="startTime"
            onChange={[MockFunction]}
            value="11:30"
          >
            <ForwardRef(MenuItem)
              value="10:30"
            >
              10:30
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="11:00"
            >
              11:00
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="11:30"
            >
              11:30
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="12:00"
            >
              12:00
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="12:30"
            >
              12:30
            </ForwardRef(MenuItem)>
          </ForwardRef(Select)>
        </ForwardRef(Box)>
        <ForwardRef(Box)
          alignItems="center"
          display="flex"
          justifyContent="space-between"
        >
          <ForwardRef(Typography)
            mr={2}
            variant="subtitle1"
          >
            ถึง
          </ForwardRef(Typography)>
          <ForwardRef(Select)
            disabled={false}
            name="endTime"
            onChange={[MockFunction]}
            value="11:30"
          >
            <ForwardRef(MenuItem)
              value="11:30"
            >
              11:30
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="12:00"
            >
              12:00
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="12:30"
            >
              12:30
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="13:00"
            >
              13:00
            </ForwardRef(MenuItem)>
            <ForwardRef(MenuItem)
              value="13:30"
            >
              13:30
            </ForwardRef(MenuItem)>
          </ForwardRef(Select)>
        </ForwardRef(Box)>
      </ForwardRef(Box)>
    }
    onClick={[Function]}
    sx={
      Object {
        "alignItems": "flex-start",
        "display": "flex",
      }
    }
  />
</ForwardRef(Stack)>
`;
